lighthouse-assert/assert/assert.ts

```diff
@@ -0,0 +1,405 @@
+        difference = new ObjectDifference(keyPath, {score: actualValue}, {score: expectedValue});
+      }
+      const subDifference = difference.getDiff();
+      if (subDifference) return diff = subDifference;
+    }
+    return diff;
+  }
+
+  /**
+   * Checks if the actual and expected object values are equal
+   *
+   * @return {boolean}
+   */
+  matchesExpectation(): boolean {
+    return Object.is(this.actual, this.expected);
+  }
+}
+
+
+class ObjectDifference implements IDifference {
+  private path: string;
+  private actual: INoneObjectActualAudit;
+  private expected: INoneObjectExpectedAudit;
+
+  /**
+   * Constructor
+   * @param {string} path
+   * @param {IActualAudit} actual
+   * @param {IExpectedAudit} expected
+   */
+  constructor(path: string, actual: INoneObjectActualAudit, expected: INoneObjectExpectedAudit) {
+    this.path = path;
+    this.actual = actual;
+    this.expected = expected;
+  }
+
+  /**
+   * Walk down expected result, comparing to actual result. If a difference is found,
+   * the path to the difference is returned, along with the expected primitive value
+   * and the value actually found at that location. If no difference is found, returns
+   * null.
+   *
+   * Only checks own enumerable properties, not object prototypes, and will loop
+   * until the stack is exhausted, so works best with simple objects (e.g. parsed JSON).
+   *
+   * @return {IDiff}
+   */
+  getDiff(): IDiff {
+    let diff: IDiff = {};
+
+    if (this.matchesExpectation()) return diff;
+
+    for (const expectationType of Object.keys(this.expected)) {
+      const expectedByType = this.expected[expectationType];
+      // If they aren't both an object we can't recurse further, so this is the difference.
+      if (this.actual.score === null || expectedByType === null ||
+          typeof this.actual.score !== 'object' || typeof expectedByType !== 'object' ||
+          expectedByType instanceof RegExp) {
+        diff = {path: this.path, actual: this.actual.score, expected: this.expected.score};
+      }
+    }
+    return diff;
+  }
+
+  /**
+   * Checks if the actual value matches the expectation. Does not recursively search. This supports
+   * greater than/less than operators, e.g. "<100", ">90"
+   *
+   * @return {boolean}
+   */
+  matchesExpectation(): boolean {
+    const actualValue = this.normalize(this.actual.score);
+    const normalizedExpected: INormalizedExpectedScore = {
+      warn: this.normalize(this.expected.score.warn),
+      error: this.normalize(this.expected.score.error)
+    };
+    return this.inRange(actualValue, normalizedExpected);
+  }
+
+  /**
+   * Checks if the actual value in warning and error range
+   *
+   * @param {number} actual
+   * @param {INormalizedExpectedScore} expected
+   * @return {boolean}
+   */
+  private inRange(actual: number, expected: INormalizedExpectedScore): boolean {
+    return actual >= expected.error || actual >= expected.warn && actual < expected.error;
```

Comment:
it seems that greater the greater than/less than operators in the error and warn are thrown away then?

Reply:
I'm not sure I got your point correct. Are you concerned that original `expected.score.warn` and `expected.score.error` are changed during normalizing?

Reply:
Well it looks like you're setting the warn and expectations as strings with `>=` or `<=` but then the actual comparison that happens here doesn't seem to use that operator information since the operators are hard-coded

Reply:
now I got it) thx